python/singa/sonnx.py

```diff
@@ -18,272 +18,359 @@
+                                             name=op.name,
+                                             kernel_shape=k,
+                                             pads=p,
+                                             strides=s))
             else:
-                if(isinstance(op,Concat)):
-                    node = [helper.make_node(curop, inputs=pre, outputs=[cur], name=name,axis=int(op.axis))] + node
-                elif(isinstance(op,_Conv2d)):
-                    pads=[op.handle.padding_h,op.handle.padding_h,op.handle.padding_h,op.handle.padding_h]
-                    stride=[op.handle.stride_h,op.handle.stride_w]
-                    k = [op.handle.kernel_h, op.handle.kernel_w]
-                    node = [helper.make_node(curop, inputs=pre, outputs=[cur], name=name, kernel_shape=k,pads=pads,strides=stride)] + node
-                elif(isinstance(op,_Pooling2d)):
-                    k = [op.handle.kernel_h, op.handle.kernel_w]
-                    s = [op.handle.stride_h, op.handle.stride_w]
-                    p = [op.handle.pad_h,op.handle.pad_h, op.handle.pad_w,op.handle.pad_w]
-                    if (op.handle.is_max_pooling):
-                        node = [helper.make_node(curop, inputs=pre, outputs=[cur], name=name,kernel_shape=k,pads=p,strides=s)] + node
-                    else:
-                        node = [helper.make_node('AveragePool', inputs=pre, outputs=[cur], name=name, kernel_shape=k,
-                                                 pads=p, strides=s)] + node
-                elif (isinstance(op, _BatchNorm2d)):
-                    pre.append(cur + 'op.running_mean')
-                    pre.append(cur + 'op.running_var')
-                    dummy0 = to_numpy(tensor.Tensor(device=op.running_mean.device(), data=op.running_mean))
-                    dummy1 = to_numpy(tensor.Tensor(device=op.running_var.device(), data=op.running_var))
-                    node = [helper.make_node(curop, inputs=pre, outputs=[cur], name=name)] + node
-
-                    node = [helper.make_node('Constant', inputs=[], outputs=[pre[3]],
-                                             value=numpy_helper.from_array(dummy0))] + node
-                    node = [helper.make_node('Constant', inputs=[], outputs=[pre[4]],
-                                             value=numpy_helper.from_array(dummy1))] + node
+                node.append(helper.make_node('AveragePool',
+                                             inputs=inputs,
+                                             outputs=outputs,
+                                             name=op.name,
+                                             kernel_shape=k,
+                                             pads=p,
+                                             strides=s))
+        elif (isinstance(op, autograd._BatchNorm2d)):
+            dummy0 = tensor.to_numpy(tensor.Tensor(device=op.running_mean.device(), data=op.running_mean))
```

Comment:
dummy tensors will be processed in line 355. no need to process them here explicitly.

Reply:
i just rechecked the code, running var and running mean will not processed in line 355
so i write some extra code here

Reply:
running var and running mean are tensors generated by Dummy. why are they missed in line 355?